app/src/androidTest/java/com/github/se/travelpouch/ui/dashboard/TimelineTest.kt

```diff
@@ -0,0 +1,83 @@
+package com.github.se.travelpouch.ui.dashboard
+
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.test.assertIsDisplayed
+import androidx.compose.ui.test.assertTextEquals
+import androidx.compose.ui.test.hasTestTag
+import androidx.compose.ui.test.junit4.createComposeRule
+import androidx.compose.ui.test.onNodeWithTag
+import com.github.se.travelpouch.model.events.Event
+import com.github.se.travelpouch.model.events.EventRepository
+import com.github.se.travelpouch.model.events.EventType
+import com.github.se.travelpouch.model.events.EventViewModel
+import com.google.firebase.Timestamp
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.`when`
+import org.mockito.kotlin.any
+
+class TimelineTest {
+
+  private lateinit var mockEventRepository: EventRepository
+  private lateinit var mockEventViewModel: EventViewModel
+
+  @get:Rule val composeTestRule = createComposeRule()
+
+  val events_test =
+      listOf(
+          Event(
+              "1",
+              EventType.NEW_DOCUMENT,
+              Timestamp(0, 0),
+              "eventTitle",
+              "eventDescription",
+              null,
+              null),
+          Event("2", EventType.START_OF_JOURNEY, Timestamp(0, 0), "it", "it", null, null),
+          Event("3", EventType.NEW_PARTICIPANT, Timestamp(0, 0), "it", "it", null, null),
+          Event("3", EventType.OTHER_EVENT, Timestamp(0, 0), "it", "it", null, null))
+
+  @Before
+  fun setUp() {
+    mockEventRepository = mock(EventRepository::class.java)
+    mockEventViewModel = EventViewModel(mockEventRepository)
+
+    // `when`(mockEventViewModel.events.value).thenReturn(events_test)
```

Comment:
Maybe you can add an explanation about how we can test and what is this line (what is events_test). It could help others understand the test and how test it.

Reply:
I don't understand the "how" part

Reply:
Yes, I'm referring to that! I'll let you add it, thanks